cowprotocol/services
# Pull S3 utils into own crate

# Description
Mostly blatant copy of [existing module](https://github.com/cowprotocol/services/blob/a98bd428ee42dbcd5980d2e5966fbe52eef4f94c/crates/solver/src/s3_instance_upload_arguments.rs) to allow reuse in the autopilot and other potentially other components once we remove the solver crate.

I'm planning to use this to get rid of the very bid auction log in the autopilot, which is making kibana slow.

# Changes
- [ ] Generalize code to work with arbitrary serializable objects

## How to test
Export AWS credentials locally, set `BUCKET=staging-bucket-133876976458` and run the unit test. See file being uploaded.

## Related Issues

#1752
